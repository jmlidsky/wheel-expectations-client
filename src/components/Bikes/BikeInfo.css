.bike-info {
    background: white;
    border-radius: 4px;
    padding: 15px 30px 30px 30px;
}

.bike-image{
    margin: 10px 0px;
    width: 100%;
}

.bike-info-name {
    color: #7E053F;
    margin: 20px 0;
    border-bottom: 3px solid #7E053F;
    padding-bottom: 20px;
    font-size: 1.4em;
    letter-spacing: 4px;
    font-weight: bold;
    text-transform: uppercase;
    text-align: center;
}

.bike-description {
    line-height: 1.5em;
}

.button-wrapper {
    display: flex;
    flex-direction: column;
}

.go-back-button {
    margin: 20px 0px;
    border-radius: 4px;
    padding: 20px;
    background: linear-gradient(180.1deg, rgba(255, 255, 255, 0) 43.72%, rgba(0, 0, 0, 0.41) 99.1%), #7E053F;
    color: white;
    font-family: 'Oswald', sans-serif;
    font-size: .9em;
    letter-spacing: 2px;
    font-weight: bold;
    text-transform: uppercase;
}

.go-back-button:hover {
    background: white;
    color: #7E053F;
}

@media (min-width: 768px) {
    .bike-info-container {
        width: 40%;
        margin: 0 auto;
    }

    .go-back-button {
        width: 50%;
        margin: 20px auto;
    }
}