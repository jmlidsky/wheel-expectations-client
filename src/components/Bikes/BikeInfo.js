import React, { Component } from 'react'
import Context from '../../Context'
import './BikeInfo.css'

class BikeInfo extends Component {
    static contextType = Context

    render() {
        const { bikes } = this.context
        const bikeId = + this.props.match.params.id
        const selectedBike = bikes.find(bike => {
            return bike.id === bikeId
        })
        
        // null check
        return selectedBike
            ? (
                <div className="bike-info-container">
                    <div className="bike-info">
                        <h4 className="bike-info-name">{selectedBike.bike_name}</h4>
                        <img src={selectedBike.bike_image} alt={selectedBike.bike_name} className="bike-image" />
                        <p className="bike-description">{selectedBike.bike_description}</p>
                    </div>
                    <div className="button-wrapper">
                        <button type="button" className="go-back-button" onClick={() => this.props.history.goBack()}>Back to Guide</button>
                    </div>
                </div>
            )
            : null
    }
}

export default BikeInfo;