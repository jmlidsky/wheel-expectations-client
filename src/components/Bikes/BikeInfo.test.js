import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import BikeInfo from './BikeInfo'

it('renders without crashing', () => {
  const props = {
    match: {
      params: {
        id: "1"
      }
    }
  }
  const context = {
    bikes: [
      {
        "id": "1",
        "bike_name": "Road",
        "bike_description": "Road bikes are designed to be ridden fast on smooth pavement. They have smooth, skinny tires and drop handlebars, and can be used for on-road racing. They are usually lighter than other types of bicycles. People find them uncomfortable and unstable on unpaved trails.",
        "bike_category": "Road",
        "bike_image": "https://www.bikeride.com/images/bike/Road-Bike-Chain-Reaction.jpg"
      },
      {
        "id": "2",
        "bike_name": "Cyclocross",
        "bike_description": "Cyclocross bikes are a special type of road bike designed to be raced on a mixed surface course (combination of pavement, unpaved trails, gravel, grass). They have a drop handlebar like regular road bikes, but the tires are a littler wider for more off-road traction, and they have a different style of brake that helps to prevent mud buildup in the frame.",
        "bike_category": "Road",
        "bike_image": "https://images.giant-bicycles.com/b_white,c_pad,h_650,q_80/spxknltjc4fqrijlkcy6/MY19-TCX-SLR-2_Color-A.jpg"
      },
      {
        "id": "3",
        "bike_name": "Touring",
        "bike_description": "Touring bikes are a special type of road bike. They are designed to be ridden on pavement. They have all of the necessary mounting bolts for cargo racks and fenders, and although they still have a drop handlebar, they usually have a more relaxed frame design so that the rider is more upright, for more comfort when riding long distances for multiple days at a time. They have a lower gear range compared to regular road bikes, to allow for carrying heavy loads up steep hills. They also make good commuter bikes, because of their durability and ability to carry heavy loads.",
        "bike_category": "Road",
        "bike_image": "https://vader-prod.s3.amazonaws.com/1551217303-kona-sutra-1551217296.jpg"
      }
    ]
  }
  
  const div = document.createElement('div');
  ReactDOM.render(
    <BrowserRouter>
      <BikeInfo selectedBike={selectedBike} />
    </BrowserRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});
