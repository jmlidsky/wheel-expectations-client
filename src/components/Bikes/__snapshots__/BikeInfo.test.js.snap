// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <BikeInfo
      match={
        Object {
          "params": Object {
            "id": "1",
          },
        }
      }
    />
  </Router>
</BrowserRouter>
`;
